packages/video_player_videohole/tizen/src/drm_license_helper.cc

```diff
@@ -0,0 +1,804 @@
+      return DRM_E_NETWORK_CURL;
+    }
+  }
+
+  return DRM_SUCCESS;
+}
+
+void HttpClose(SHttpSession* http_session) {
+  if (!http_session) {
+    return;
+  }
+
+  if (http_session->curl_handle != nullptr) {
+    curl_easy_cleanup(http_session->curl_handle);
+  }
+
+  if (http_session->post_data) {
+    free(http_session->post_data);
+  }
+
+  if (http_session->body.data) {
+    free(http_session->body.data);
+  }
+
+  if (http_session->header.data) {
+    free(http_session->header.data);
+  }
+
+  free(http_session);
+}
+
+size_t ReceiveHeader(void* ptr, size_t size, size_t nmemb, void* stream) {
+  LOG_DEBUG("[DrmLicenseHelper] size:%d nmemb:%d", size, nmemb);
+
+  size_t data_size = size * nmemb;
+
+  if (data_size > 0) {
+    SHttpSession* http_session = reinterpret_cast<SHttpSession*>(stream);
+
+    if (!AppendData(&http_session->header, ptr, data_size)) {
+      return 0;
+    }
+  }
+  return data_size;
+}
+
+size_t ReceiveBody(void* ptr, size_t size, size_t nmemb, void* stream) {
+  LOG_DEBUG("[DrmLicenseHelper] size:%d nmemb:%d", size, nmemb);
+
+  size_t data_size = size * nmemb;
+
+  if (data_size > 0) {
+    SHttpSession* http_session = reinterpret_cast<SHttpSession*>(stream);
+
+    if (!AppendData(&http_session->body, ptr, data_size)) {
+      return 0;
+    }
+  }
+  return data_size;
+}
+
+size_t SendBody(void* ptr, size_t size, size_t nmemb, void* stream) {
+  LOG_DEBUG("[DrmLicenseHelper] size:%d nmemb:%d", size, nmemb);
+
+  SHttpSession* http_session = reinterpret_cast<SHttpSession*>(stream);
+
+  size_t avail_data = http_session->post_data_len - http_session->send_data_len;
+  size_t can_send = size * nmemb;
+
+  if (avail_data == 0) {
+    return 0;
+  }
+
+  if (can_send > avail_data) {
+    can_send = avail_data;
+  }
+
+  memcpy(ptr, http_session->post_data + http_session->send_data_len, can_send);
+  http_session->send_data_len += can_send;
+
+  return can_send;
+}
+
+}  // namespace
+
+DRM_RESULT DrmLicenseHelper::DoTransactionTZ(
+    const char* http_server_url, const void* challenge,
+    unsigned long challenge_len, unsigned char** response,
+    unsigned long* response_len, DrmType type, const char* http_cookie,
+    SExtensionCtxTZ* http_ext_ctx) {
+  *response = nullptr;
+  *response_len = 0;
+
+  const char* http_url = http_server_url;
+  SHttpSession* http_session;
+  char* redirect_url = nullptr;
+
+  DRM_RESULT drm_result = DRM_SUCCESS;
+
+  // Redirection 3 times..
+  for (int i = 0; i < 3; i++) {
+    if (!(http_session = HttpOpen())) {
+      LOG_ERROR("[DrmLicenseHelper] Failed to open HTTP session.");
+      break;
+    }
+
+    char* soap_header = nullptr;
+    char* http_header = nullptr;
+    char* user_agent = nullptr;
+    bool* cancel_request = nullptr;
+
+    if (http_ext_ctx != nullptr) {
+      if (http_ext_ctx->http_soap_header) {
+        soap_header = http_ext_ctx->http_soap_header;
+      }
+
+      if (http_ext_ctx->http_header) {
+        soap_header = http_ext_ctx->http_header;
```

Comment:
Maybe,  Is the below correct?

```suggestion
        http_header = http_ext_ctx->http_header;
```

Reply:
Nice catch